File: ios/Debug.m
#import "Debug.h"

@implementation Debug (ObjC)

+(void)logWithFormat:(NSString*)format, ... {
  NSMutableArray<id> *args = [[NSMutableArray alloc] init];
  va_list varargs;
  va_start(varargs, format);
  for (id arg = va_arg(varargs, id); arg != nil; arg = va_arg(varargs, id)) {
    [args addObject:arg];
  }
  va_end(varargs);
  [self logWithFormat:format arguments:args];
}

@end

File: ios/Debug.swift
import Foundation

@objc
class Debug: NSObject {
  @objc
  public static func log(message: String) {
    print(message)
  }

  public static func log(format: String, _ args: CVarArg...) {
    print(String(format: format, arguments: args))
  }

  @objc
  public static func log(format: String, arguments args: [Any]) {
    print(String(format: format, arguments: args as! [CVarArg]))
  }

  @objc
  static func log(error: Error) {
    print(error.localizedDescription)
  }
}
